[scs_internal] Avoid out-of-bounds read on zero-length array

We should report this bug to upstream.

--- src/linalg.c
+++ src/linalg.c
@@ -153,6 +153,9 @@
 */
 
 scs_float SCS(norm_inf)(const scs_float *a, scs_int len) {
+  if (len <= 0) {
+    return 0;  // Follow the semantics of BLASI(lange) for zero-size array.
+  }
   blas_int bone = 1;
   blas_int blen = (blas_int)len;
   scs_int idx = (scs_int)BLASI(amax)(&blen, a, &bone);
